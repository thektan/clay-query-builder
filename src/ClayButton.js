import React from 'react';
import {PropTypes} from 'prop-types';

class ClayButton extends React.Component {
    render() {
        const {label, type} = this.props;

        return (
            <button className="btn btn-secondary" {...this.props}>{label}</button>
        );
    }
}

ClayButton.propTypes = {
    label: PropTypes.string,
    type: PropTypes.string
}

export default ClayButton;