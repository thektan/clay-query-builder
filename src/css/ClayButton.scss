.icon {
	margin: 0;
}

.text-icon {
	margin-left: 8px;
}