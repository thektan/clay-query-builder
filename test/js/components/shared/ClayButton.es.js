import React from 'react';
import ClayButton from 'components/shared/ClayButton.es';
import {cleanup, render} from 'react-testing-library';

describe(
	'ClayButton',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const {asFragment} = render(
					<ClayButton />
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);

		it(
			'should render with a label',
			() => {
				const {container} = render(
					<ClayButton label="test" />
				);

				const button = container.querySelector('button');

				expect(button.textContent).toEqual('test');
			}
		);
	}
);