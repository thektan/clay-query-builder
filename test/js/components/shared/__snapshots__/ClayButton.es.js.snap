// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayButton should render 1`] = `
<DocumentFragment>
  <button
    class="btn btn-secondary"
    type="button"
  />
</DocumentFragment>
`;
